### DISCUSSION TITLE
CockroachDB with random UUID generation

### DISCUSSION THREAD

**Discussion by idc77** [ent/ent#2356] [Q&A] [answered]
At [CockroachDB's UUID docs](https://www.cockroachlabs.com/docs/stable/uuid.html#create-a-table-with-auto-generated-unique-row-ids) you can see
```
CREATE TABLE users (
        id UUID NOT NULL DEFAULT gen_random_uuid(),
);
```

CockroachDB is Postgresql compatible. You can use pg or pgx.

Anyhow I wonder how can this be expressed for the ID field in the schema.

**Comment by idc77** [comment]
*tumbleweed*
Why do you have discussions when no one participates?

**Comment by a8m** [comment, marked as answer]
Hey @idc77!

You can do it using the entsql annotation (see [docs](https://entgo.io/docs/schema-fields#default-values)). There's actually an example for it in our repository - https://github.com/ent/ent/blob/master/entc/integration/customid/ent/schema/blob.go#L26-L28

```go
func (Blob) Fields() []ent.Field {
	return []ent.Field{
		field.UUID("id", uuid.UUID{}).
			Default(uuid.New).
			Annotations(entsql.Annotation{
				Default: "uuid_generate_v4()",
			}).
			Unique(),
		// ...
	}
}
```
